pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/db/SomaticResultUpload.java

```diff
@@ -0,0 +1,411 @@
+        }
+        if (documents.size() > 1) {
+            throw new RuntimeException(
+                    String.format("Error getting somatic file!  More than one document found for bucket %s and blobPath %s",
+                            bucket, blobPath));
+        }
+        if (documents.isEmpty()) {
+            throw new RuntimeException(String.format("Bad request for document with %s %s", bucket, blobPath));
+        }
+        return documents.get(0);
+    }
+
+    private static void runSelect(List<SomaticResultUpload> documents, PreparedStatement stmt) throws SQLException {
+        try (ResultSet rs = stmt.executeQuery()) {
+            while (rs.next()) {
+                documents.add(getResult(rs));
+            }
+        }
+    }
+
+    private static SomaticResultUpload getResult(ResultSet rs) throws SQLException {
+        return new SomaticResultUpload(
+            rs.getLong(DBConstants.SOMATIC_DOCUMENTS_PK),
+            rs.getInt(DBConstants.DDP_INSTANCE_ID),
+            rs.getString(DBConstants.DDP_PARTICIPANT_ID),
+            rs.getLong(DBConstants.PARTICIPANT_ID),
+            rs.getString(DBConstants.FILE_NAME),
+            rs.getString(DBConstants.MIME_TYPE),
+            rs.getString(DBConstants.BUCKET),
+            rs.getString(DBConstants.BLOB_PATH),
+            rs.getLong(DBConstants.CREATED_BY_USER_ID),
+            rs.getLong(DBConstants.CREATED_AT),
+            rs.getLong(DBConstants.DELETED_BY_USER_ID),
+            rs.getLong(DBConstants.DELETED_AT),
+            rs.getBoolean((DBConstants.IS_VIRUS_FREE)));
+    }
+
+    public static SomaticResultUpload createFileUpload(String realm, String ddpParticipantId, String fileName,
+                                                       String mimeType, String bucket, String blobPath, long createdByUserId) {
+        SimpleResult results = inTransaction(conn -> {
+            SimpleResult dbVals = new SimpleResult();
+            try (PreparedStatement stmt = conn.prepareStatement(SQL_INSERT_SOMATIC_DOCUMENT, Statement.RETURN_GENERATED_KEYS)) {
+                stmt.setString(1, realm);
+                stmt.setString(2, fileName);
+                stmt.setString(3, mimeType);
+                stmt.setString(4, bucket);
+                stmt.setString(5, blobPath);
+                stmt.setLong(6, createdByUserId);
+                stmt.setLong(7, Instant.now().getEpochSecond());
+                stmt.setString(8, ddpParticipantId);
+                stmt.setString(9, realm);
+                int result = stmt.executeUpdate();
+                if (result == 1) {
+                    try (ResultSet rs = stmt.getGeneratedKeys()) {
+                        if (rs.next()) {
+                            dbVals.resultValue = rs.getInt(1);
+                        }
+                    } catch (Exception e) {
+                        throw new RuntimeException("Error inserting somatic file", e);
```

Comment:
Are we handling none unique names with this exception as here?

Reply:
We're making the files in GCS unique with the blobPath.  This is the metadata out of the file that was uploaded.  See SomaticResultUploadService line 107 for how we're building the blobPath.

Reply:
...and I forgot to mention that there's a UNIQUE constraint on bucketName+blobPath in pepper-apis/dsm-core/src/main/resources/liquibase/PEPPER-811_somatic_documents.xml